apollo-router/src/axum_factory/compression/mod.rs

```diff
@@ -79,7 +79,8 @@ where {
                         }
                     }
                     Ok(data) => {
-                        let mut buf = BytesMut::zeroed(1024);
+                        // most compression algorithms have a compression ratio of more than 90% for JSON
+                        let mut buf = BytesMut::zeroed(data.len() / 10);
```

Comment:
Not this change, but does it need to be zeroed or could we just create `with_capacity()`?

Reply:
the compressor has to write into a mutable slice, it has to be initialized. I could take a look at MaybeUninit here, but I don't think the PartialBuffer type is set up for it